src/main/scala/temple/generate/target/openapi/OpenAPIType.scala

```diff
@@ -0,0 +1,33 @@
+package temple.generate.target.openapi
+
+import io.circe.Json
+import io.circe.syntax._
+import temple.generate.target.openapi.OpenAPIType.{OpenAPIArray, OpenAPIObject}
+import temple.utils.MonadUtils.MatchPartial
+
+import scala.collection.immutable.ListMap
+
+sealed abstract private[openapi] class OpenAPIType(fieldEntries: Seq[(String, Json)]) extends Jsonable {
+  def typeString: String
+  final lazy val customFields: Map[String, Json] = fieldEntries.to(ListMap)
+
+  override def toJsonMap: Map[String, Option[Json]] =
+    ListMap(
+      "type"       -> Some(typeString.asJson),
+      "properties" -> this.matchPartial { case openAPIType: OpenAPIObject => openAPIType.properties.asJson },
```

Comment:
Just a thought - is this good design, for the abstract "parent" type to know about the "children" types?

Reply:
How does this alternative look? Removes the need for matchPartial, as each child passes up the additional fields they define
https://github.com/TempleEight/temple/compare/openapi-gen...openapi-gen-rewrite-openapitype-tojsonmap-2

Another alternative, but personally I think this seems a little more repetitive:
https://github.com/TempleEight/temple/compare/openapi-gen...openapi-gen-rewrite-openapitype-tojsonmap